Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -594,31 +595,33 @@ def early_shutdown(self):
         """ Called as soon as Session shutdown is initiated. Used to start
         shutdown tasks that takes some time and that can run in parallel
         to checkpointing, etc.
+        :returns a Deferred that will fire once all dependencies acknowledge they have shutdown.
         """
         self._logger.info("tlm: early_shutdown")
 
         self.cancel_all_pending_tasks()
 
         # Note: sesslock not held
+        wait_defer_list = []
         self.shutdownstarttime = timemod.time()
         if self.torrent_checker:
-            self.torrent_checker.shutdown()
+            wait_defer_list.append(self.torrent_checker.shutdown())
```

Comment:
This still works because the shutdown methods are still blocking, but once they become async, this will break as the module shutdown order won't be guaranteed anymore.

Right now I can't think of a not too involved way of doing this that doesn't involve using @inlineCallback until we have a smarter system.